components/formats-bsd/src/loci/formats/in/OBFReader.java

```diff
@@ -561,131 +651,353 @@ private String readString() throws IOException {
     }
   }
 
-  @Override
-  public byte[] openBytes(int no, byte[] buffer, int x, int y, int w, int h)
-    throws FormatException, IOException 
+  private long remainingBytesInChunk(Stack stack)
+    throws IOException
   {
-    FormatTools.checkPlaneParameters(this, no, buffer.length, x, y, w, h);
+    return state.chunkFileStart +
+        state.chunkSize -
+        in.getFilePointer();
+  }
 
-    final int rows = getSizeY();
-    final int columns = getSizeX();
-    final int bytesPerPixel = getBitsPerPixel() / 8;
+  private void readFromStackRaw(Stack stack, byte[] buffer, int bufferOffset, int bytes)
+    throws IOException, FormatException
+  {
+    long remainingBytesInChunk = remainingBytesInChunk(stack);
```

Comment:
Might be worth explicitly checking and throwing an exception if ```remainingBytesInChunk``` is negative?

Reply:
In theory it shouldn't happen (unless the writer wrote an incorrect file, the file corrupted or the reader is incorrect). I added a `FormatException` which I hope is what you had in mind.